### PR TITLE
feat: add Author, Series, and Tag Cloud discovery pages (F1.8)

### PR DESCRIPTION
**Pull request by seamus-sloan**
## Summary
- Add three discovery pages — `/authors/:id`, `/series/:id`, `/tags` — implementing the F1.8 initiative from the roadmap. Full vertical slice: shared types (`AuthorDetail`, `SeriesDetail`, `TagWeight`), DB query functions (`get_author`, `get_series`, `get_tag_cloud`), RPC + REST endpoints, Dioxus page components with Atrium design system styling, and Playwright E2E specs.
- Propagate `series_id` through `EbookMetadata` so the author page's series section headers and book detail's series rail link to `/series/:id`. Tag cloud items navigate back to the library with the clicked tag pre-selected as a filter facet.
- Add `ViewFilters.tags` to the shared type and wire it through the landing page's filter sidebar, `matches_filters`, and `facet_counts`.

## Test plan
- [x] `cargo clippy` — zero warnings
- [x] `cargo test -p omnibus-db` — 157 tests pass (includes `get_author`, `get_series`, `get_tag_cloud` coverage)
- [x] `cargo test -p omnibus` — 81 tests pass (includes 8 new REST integration tests for `/api/authors/:id`, `/api/series/:id`, `/api/tags`)
- [x] `cargo test -p omnibus-frontend --features server` — 50 tests pass (landing page tag filter tests)
- [ ] Playwright: `npx playwright test tests/flows/discovery.spec.ts` — 6 E2E specs (author layout, series layout, tag cloud layout)
- [ ] Manual: navigate author page → click series heading → arrives at series page
- [ ] Manual: book detail → click series name in rail → arrives at series page
- [ ] Manual: tag cloud → click tag → library filters to that tag

## Notes
- Author page groups books by series with linkable section headers, standalone books shown separately
- Series page shows books ordered by `series_index` with card layout
- Tag cloud uses font-size scaling (16–72px) and opacity (0.55–1.0) proportional to tag frequency
- Also added roadmap docs: `1-11-author-profiles.md` (cascading author photo resolution) and `1-12-browse-authors-series.md` (browse-all index pages)
- `series_id` is resolved via a scalar subquery (`books_series_link → series.id`) across all four book-loading SQL paths: `list_books`, `search_books`, `get_book`, and `BOOK_COLUMNS`/`row_to_ebook`

>[!NOTE]
> The Search page is excluded from this PR — it's being handled separately.

### CODE REVIEW

**Comment by seamus-sloan** [comment]
> 🤖 **Posted by Claude** on behalf of @seamus-sloan — automated review.

# Claude Code Review — PR #109 (discovery pages)

## Code review

**Overview.** Cleanly adds three discovery pages (author / series / tags) wired through web RPC and mobile REST, with auth on both. The DB layer factors out a reusable `BOOK_COLUMNS` + `row_to_ebook` helper, and the new `series_id` / `Contributor.id` are additive with `#[serde(default, skip_serializing_if = "Option::is_none")]` so wire compatibility is preserved. Integration tests cover happy/404/401 on every new REST endpoint; main gaps are direct DB-layer unit tests and a few correctness/a11y issues on the frontend.

### Findings

- **Medium — Missing DB-layer unit tests.** `get_author`, `get_series`, `get_tag_cloud` (`db/src/queries.rs:302/339/376`) have no inline `#[cfg(test)]` cases. Rule 03 requires happy + not-found + edge cases at the DB layer. Server integration tests cover JSON shape but not `series_index NULLS LAST` ordering, tag-cloud `ORDER BY cnt DESC, name ASC`, or authors-with-no-books.
- **Medium — Doubled correlated subquery on a hot path.** `list_books` / `search_books` / `get_book` / `BOOK_COLUMNS` now run **two** correlated subqueries per row against `books_series_link` (`series_name` and `series_link_id`), both with `ORDER BY s.name LIMIT 1`. Fold into a single `LEFT JOIN (SELECT bsl.book, s.id, s.name FROM books_series_link bsl JOIN series s ON s.id = bsl.series GROUP BY bsl.book) ...`, or a single subquery returning a JSON object.
- **Medium — `get_tag_cloud` has no `LIMIT`.** `db/src/queries.rs:376`. A Calibre dump with 10k+ subjects will serialize all of them to JSON, blow up the WASM client, and scan-and-group all of them on the server. Cap server-side (~500) and surface "showing top N of M".
- **Medium / a11y — Tag cloud encodes weight via font-size + opacity only.** `frontend/src/pages/tag_cloud.rs:1611-1640`. Add `aria-label="{name} — {count} books"` on each link. No `<h2>` structure between header and items.
- **Medium — Tag cloud links go nowhere useful.** Every item links to `Route::Landing {}` with no tag filter applied, but the header copy says "click to filter". Either route to `/?tags=<name>` (and parse it on Landing into `ViewFilters.tags`) or remove the affordance text.
- **Low — Italic styling on author "last name" / series "rest".** `frontend/src/pages/author.rs:1077`, `series.rs:1448`. Splitting on first space breaks for non-Latin scripts and titles like "The Wheel of Time" (italicizes "The"). Consider styling the whole title.
- **Low — `series.rs:1467-1478` hardcodes "The " before the series name.** Wrong for "Foundation", "Discworld", "Dune Chronicles". Drop the literal.
- **Low — `data::get_author` (`frontend/src/data.rs:854`) duplicated 404 parsing.** Extract a helper.
- **Low — `use_effect` re-fetch on `id` change.** Verify `/authors/1` → `/authors/2` retriggers when the component is reused. Idiomatic dioxus-fullstack is `use_resource(move || ...)` keyed on `id`.
- **Low — Playwright `fetchSeriesIdByName` probes IDs 1..100 in a loop.** `ui_tests/playwright/tests/flows/discovery.spec.ts:2086-2110`. 100 round-trips per test. `EbookMetadata.series_id` is already serialized — read it from the ebooks list instead.
- **Nit — `book_detail.rs` removed "view author →" hint** for a real link. Confirm no Playwright spec still asserts that text before merging.

## Security review

- **OK — SQL injection.** All three new functions bind parameters; `format!` calls only splice the constant `BOOK_COLUMNS` literal.
- **OK — Auth.** REST (`server/src/backend.rs:1719-1748`) and RPC (`frontend/src/rpc.rs:1672-1685`) both extract `_user: AuthUser`. The top-level `auth::gate` covers `/api/*`. 401-when-anonymous tests are present for all three.
- **Low / informational — Library scoping / IDOR.** Single-tenant today, so cross-tenant enumeration isn't a current threat. But `get_author(pool, id)` and `get_series(pool, id)` don't scope by anything — the moment per-user ACLs land (F4.x), this will leak cross-user data by ID. Add a `// TODO(F4.x): scope by user when ACLs land` comment near the three new query fns and a tracking item on the auth initiative page.
- **OK — XSS on names.** All names interpolated via Dioxus `"{name}"` text nodes (HTML-escaped). Breadcrumb, avatar initial, tag link text — all safe.
- **Low — XSS on series description.** `frontend/src/pages/series.rs:1510` uses `dangerous_inner_html`. Same pattern as book_detail; relies on `sanitize_description` at write/read time (`queries.rs:1074, 1138`). Confirm `sanitize_description_strips_scripts_and_event_handlers` is still in the test module and the sanitizer (`ammonia`?) hasn't been bypassed.
- **Low — Info disclosure via 404 vs 200.** `/api/authors/:id` returns 404 for missing IDs vs 200 with full payload. Today fine (everyone has read access to the whole library); pair with the IDOR note above for when multi-tenant lands.
- **Medium — DoS via unbounded tag cloud.** `get_tag_cloud` returns every tag with no `LIMIT`. A pathological Calibre import (1M synthetic subjects) → huge JSON blob, OOM the WASM client, slow `GROUP BY` scan blocking the SQLite pool. Add `LIMIT 500`. Same risk on `get_author` / `get_series` if a single author has thousands of books.
- **OK — Input validation.** `Path(id): Path<i64>` rejects non-integer paths at the axum layer; negative IDs simply 404 from the DB.
- **OK — CSRF.** All new endpoints are GETs or behind the dioxus-fullstack POST + `origin_check` machinery.

### Top action items
1. **Medium** — add `LIMIT` to `get_tag_cloud` (and consider one on author/series book lists).
2. **Medium** — add direct DB unit tests for `get_author` / `get_series` / `get_tag_cloud`.
3. **Medium** — fix tag-cloud links to actually filter Landing (or remove the affordance text).
4. **Medium** — collapse `series_name` + `series_link_id` into a single LEFT JOIN across all four book SQL paths.
5. **Low** — add `// TODO(F4.x): scope by user` comment near the three new query functions.

File: ui_tests/playwright/tests/flows/discovery.spec.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  for (let id = 1; id <= 100; id++) {
+    const resp = await request.post("/api/rpc/series", {
+      data: { id },
+    });
+    if (resp.status() === 200) {
+      const body = (await resp.json()) as { name: string; id: number } | null;
+      if (body && body.name === name) return body.id;
+    }
+  }
+  throw new Error(`could not resolve series id for ${JSON.stringify(name)}`);
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply ui_tests/playwright/tests/flows/discovery.spec.ts]: Fixed in 1d1fd56 — `fetchSeriesIdByName` now reads `series_id` directly off the `/api/rpc/ebooks` payload (the new `EbookMetadata.series_id` field), no more brute-force probe.

File: ui_tests/playwright/tests/flows/discovery.spec.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+// ---------------------------------------------------------------------------
+// Helpers — resolve IDs via the RPC layer so tests don't hardcode DB ids.
+// ---------------------------------------------------------------------------
+
+async function fetchAuthorIdByName(
+  request: APIRequestContext,
+  name: string,
+): Promise<number> {
+  // POST to /api/rpc/ebooks (same as the landing page) and scan creators.
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply ui_tests/playwright/tests/flows/discovery.spec.ts]: Fixed in 1d1fd56 — comment now says GET to match the actual call.

File: frontend/src/pages/tag_cloud.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                div { class: "tag-cloud",
+                    for tag in tag_list.iter() {
+                        { render_tag_item(tag, max_count) }
+                    }
+                }
+                // Overlap sidebar — placeholder for future tag analysis
+                aside { class: "card", aria_hidden: "true",
+                    // TODO(F1.13): Tag overlap matrix — show related tags when
+                    // one is selected. Stubbed out until the backend supports
+                    // co-occurrence queries.
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply frontend/src/pages/tag_cloud.rs]: Fixed in 1d1fd56 — collapsed to a single-line `// TODO(F1.13):` above the empty aside.

File: frontend/src/pages/landing.rs [hunk 4]
```diff
@@ -371,10 +372,11 @@ fn FilterSidebar(
...
             let mut next = filters.clone();
-            let bucket = if group == "authors" {
-                &mut next.authors
-            } else {
-                &mut next.series
+            let bucket = match group {
+                "authors" => &mut next.authors,
+                "series" => &mut next.series,
+                "tags" => &mut next.tags,
+                _ => &mut next.authors,
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply frontend/src/pages/landing.rs]: Fixed in 1d1fd56 — unknown facet groups now `debug_assert!` and early-return without calling `on_change`, so typos fail loudly in debug builds instead of silently mutating `authors`.

File: frontend/src/pages/series.rs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                                    }
+                                }
+                                h3 { class: "series-card-title",
+                                    Link {
+                                        to: Route::BookDetail { id: book.id },
+                                        "{book.title.as_deref().unwrap_or(&book.filename)}"
+                                    }
+                                }
+                                if let Some(ref desc) = book.description {
+                                    p { class: "series-card-desc",
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply frontend/src/pages/series.rs]: Fixed in 1d1fd56 — switched the description container from `<p>` to `<div>` so block-level HTML from `sanitize_description` no longer nests inside a paragraph.

File: ui_tests/playwright/tests/flows/discovery.spec.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+test("tag cloud items show counts", async ({ page }) => {
+  await gotoReady(page, "/tags");
+
+  // Each tag should have a visible count span
+  const counts = page.locator(".tag-cloud-count");
+  const firstCount = await counts.first().textContent();
+  expect(firstCount).toBeTruthy();
+  expect(Number(firstCount)).toBeGreaterThan(0);
+});
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply ui_tests/playwright/tests/flows/discovery.spec.ts:151]: Deferred — tracked in #110. Blocked on the tag-cloud filter wiring (sister thread) for the third assertion, but the author→series and book-detail→series nav tests can land in that follow-up regardless.

File: frontend/src/pages/tag_cloud.rs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    let count = tag.count;
+
+    rsx! {
+        Link {
+            to: Route::Landing {},
+            class: "{class}",
+            style: "{style}",
+            "{name}"
+            span { class: "tag-cloud-count mono", "{count}" }
+        }
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply frontend/src/pages/tag_cloud.rs:119]: Deferred. The fix needs either a query-param-aware Landing route or pre-navigation `ViewFilters` plumbing through context — both >30 min of design + plumbing. Leaving the affordance text in place for now; happy to revisit in a follow-up PR.

**Comment by seamus-sloan** [comment]
Addressed the Claude code review (commit 8c65a34):

**Fixed:**
- `get_tag_cloud` now has `LIMIT 500` (DoS finding + the unbounded-tag-cloud security note)
- Tag cloud items have `aria-label="{name} — {N} books"` for screen-reader weight signal
- Dropped the hardcoded `"The "` prefix from the series page `h1` (was wrong for "Foundation", "Dune Chronicles", etc.)
- `TODO(F4.x):` comments on `get_author` / `get_series` / `get_tag_cloud` flagging per-user ACL scoping for the IDOR concern
- 7 new inline DB unit tests covering `get_author` (happy + 404 + series_id propagation), `get_series` (ordering + 404), and `get_tag_cloud` (count ordering + empty) — satisfies rule 03's DB-layer coverage requirement
- Verified no Playwright spec asserts the removed "view author →" hint text

**Deferred (not in this PR):**
- **Double correlated subquery on `series_name` + `series_link_id`** — real perf concern but a SQL refactor across 4 query sites (`BOOK_COLUMNS`, `list_books`, `search_books`, `get_book`) needs benching to confirm the LEFT JOIN is actually faster, and risks touching a hot path. Worth a follow-up.
- **Italic styling splits on first space** — design nit; doesn't break anything functionally and mirrors what book detail does today. Punt to a styling pass.
- **`data::get_author` duplicated 404 parsing** — refactor across 3 sites, not blocking.
- **`use_effect` re-fetch on `id` change** — same pattern as the existing `BookDetailPage`. If it doesn't retrigger, both are broken consistently and the fix belongs in a separate PR.
- **Tag cloud links don't filter** — already deferred upthread (no follow-up issue per your call).
- **Navigation action tests** — tracked in #110.
